uberfire-api/src/main/java/org/uberfire/workbench/type/AnyResourceTypeDefinition.java

```diff
@@ -17,9 +17,20 @@
 package org.uberfire.workbench.type;
 
 import org.uberfire.backend.vfs.Path;
+import org.uberfire.workbench.category.Category;
+import org.uberfire.workbench.category.Others;
 
 public class AnyResourceTypeDefinition implements ResourceTypeDefinition {
 
+    private Category category;
+
+    public AnyResourceTypeDefinition() {
```

Comment:
This classes need to be injected because are part of CDI and I have to find those classes to cache them. Also the categories needs to be "plug and play", so you need to find every class injecting Any<Category>. That was the initial premise to build this part. They can't be null as you said but they have to be injected.